applications/StructuralMechanicsApplication/custom_elements/nodal_concentrated_element.cpp

```diff
@@ -432,6 +433,65 @@ void NodalConcentratedElement::CalculateDampingMatrix(
     KRATOS_CATCH( "" );
 }
 
+
+void NodalConcentratedElement::AddExplicitContribution(
+    const VectorType& rRHSVector,
+    const Variable<VectorType>& rRHSVariable,
+    const Variable<double >& rDestinationVariable,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY;
+
+    if (rDestinationVariable == NODAL_MASS) {
+        double& r_nodal_mass = GetGeometry()[0].GetValue(NODAL_MASS);
+        #pragma omp atomic
+        r_nodal_mass += GetValue(NODAL_MASS);
+    }
+
+    KRATOS_CATCH("")
+}
+
+void NodalConcentratedElement::AddExplicitContribution(
+    const VectorType& rRHSVector, const Variable<VectorType>& rRHSVariable,
+    const Variable<array_1d<double, 3>>& rDestinationVariable,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY;
+
+    auto& r_geom = GetGeometry();
+    const SizeType dimension = r_geom.WorkingSpaceDimension();
+    const SizeType number_of_nodes = r_geom.size();
+    const SizeType local_size = dimension*number_of_nodes;
+
+    if (rRHSVariable == RESIDUAL_VECTOR && rDestinationVariable == FORCE_RESIDUAL) {
+
+        Vector damping_residual_contribution = ZeroVector(local_size);
+        Vector current_nodal_velocities = ZeroVector(local_size);
+        GetFirstDerivativesVector(current_nodal_velocities);
+        Matrix damping_matrix;
+        ProcessInfo temp_process_information; // cant pass const ProcessInfo
+        CalculateDampingMatrix(damping_matrix, temp_process_information);
+        // current residual contribution due to damping
+        noalias(damping_residual_contribution) = prod(damping_matrix, current_nodal_velocities);
+
+        for (SizeType i = 0; i < number_of_nodes; ++i) {
+            SizeType index = dimension * i;
+            array_1d<double, 3>& r_force_residual = GetGeometry()[i].FastGetSolutionStepValue(FORCE_RESIDUAL);
+            for (size_t j = 0; j < dimension; ++j) {
+                #pragma omp atomic
+                r_force_residual[j] += rRHSVector[index + j] - damping_residual_contribution[index + j];
+            }
+        }
+    } else if (rDestinationVariable == NODAL_INERTIA) {
+        double& r_nodal_mass = GetGeometry()[0].GetValue(NODAL_MASS);
+        #pragma omp atomic
+        r_nodal_mass += GetValue(NODAL_MASS);
```

Comment:
with C++17 you could do the following to be safe:
```suggestion
        r_nodal_mass += std::as_const(*this).GetValue(NODAL_MASS);
```

ref: https://en.cppreference.com/w/cpp/utility/as_const